### DISCUSSION TITLE
How to use dynamic locales coming from the backend?

### DISCUSSION THREAD

**Discussion by martinszeltins** [js-fns/date-fns#3071] [General]
My locale is coming from the backend and I do not know what it will be in advance. So I cannot statically import a specific locale from `date-fns` and I don't want to import them all either. Is there a workaround for this?

For example, from the backend server I receive this locale as string

```js
let locale = await settingsApi.getSettings()
let locale = 'en' // <-- I don't know what locale backend will give me here
```

How can I now use this `locale` variable for `date-fns`?

```js
import { formatWithOptions } from 'date-fns/fp'

// Well, I can't import en because I don't know the locale in advance
import { en } from 'date-fns/locale'

function formatDate(date) {
  return formatWithOptions({ locale: en }, 'D MMMM YYYY')
}
```

**Comment by Jules010209** [comment]
Hello, how that's can be returned? I speak about you're locales from your API. Do you have a list of all the possibilities?
In fact, if the variables return the same variable name as date-fns, you can easily adapt them to your code.

**Comment by joaopedrodcf** [comment]
You can dynamic import the locale something like this ( v2 ):

```tsx
import { setDefaultOptions } from 'date-fns';
...

export const loadDateFnsLocale = async (locale: Locale) => {
    return await import(
        `date-fns/locale/${localeToDateFnsLocale[locale]}/index.js`
    );
};

export const useDateFnsLocale = () => {
   const {data} = loadDateFnsLocale('en-US'); // Example

    if (data) {
        setDefaultOptions({ locale: data.default });
    }

   ...
};
```
